exonum-java-binding-core/src/main/java/com/exonum/binding/blockchain/Blockchain.java

```diff
@@ -81,4 +89,114 @@ public long getHeight() {
     return schema.getBlockTransactions(height);
   }
 
+  /**
+   * Returns a proof list of transaction hashes committed in the block with given id.
+   *
+   * @param blockId id of the block
+   * @return a proof list of transaction hashes committed in the block.
+   *        Or null if the block with given id doesn't exist
+   */
+  @Nullable
+  public ProofListIndexProxy<HashCode> getBlockTransactions(HashCode blockId) {
+    MapIndex<HashCode, Block> blocks = schema.getBlocks();
+    Block block = blocks.get(blockId);
+    if (block == null) {
+      return null;
```

Comment:
Yes, I agree that null seems bad in this case. I think we should use checked exceptions here then? Empty list seems not okay, as we can probably have a block without transactions?

Reply:
It might make sense to delay these changes (optionals + exceptions) and ponder about various options for a while. We may discuss it during the grooming.